sklearn/utils/estimator_checks.py

```diff
@@ -314,7 +336,7 @@ def set_testing_parameters(estimator):
         # of components of the random matrix projection will be probably
         # greater than the number of features.
         # So we impose a smaller number (avoid "auto" mode)
-        estimator.set_params(n_components=1)
+        estimator.set_params(n_components=2)
```

Comment:
One way around this is to set the args this way within our internal common tests. So we wouldn't do any resetting inside check_estimator, just inside the code that calls check_estimator.

This way, when we review a scikit-learn contrib project, we can see at a glance if they need to do any weird juggling of parameters when calling check_estimator or not.

I think the only reason we were overriding n_components, max_iter, etc is to make our own test suite as fast as possible. In `polylearn` for instance, 15 iters were insufficient to reach good accuracy for some of the models, leading to check_estimator failures, so I would like a way to override it.